src/traceViewer.ts

```diff
@@ -17,67 +17,79 @@
 import { ChildProcess, spawn } from 'child_process';
 import type { TestConfig } from './playwrightTestTypes';
 import { SettingsModel } from './settingsModel';
-import { findNode } from './utils';
+import { findNode, getNonce } from './utils';
 import * as vscodeTypes from './vscodeTypes';
+import { DisposableBase } from './disposableBase';
 
-export class TraceViewer implements vscodeTypes.Disposable {
+function getPath(uriOrPath: string | vscodeTypes.Uri) {
+  return typeof uriOrPath === 'string' ?
+    uriOrPath :
+    uriOrPath.scheme === 'file' ?
+      uriOrPath.fsPath :
+      uriOrPath.toString();
+}
+
+export type TraceViewer = SpawnTraceViewer | EmbeddedTraceViewer;
+
+export class SpawnTraceViewer extends DisposableBase {
```

Comment:
Let's put these into separate files. When splitting the change, we can land SpawnTraceViewer in one PR and follow up with adding EmbeddedTraceViewer. That way we would see if there is something funny with the chosen modularity.